### ISSUE TITLE
Not able to ping between linked containers when networkPolicyRule is set to "allow linked"

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#7935]
Rancher server version - Build from master

Steps to reproduce the problem:
Have defaulNetworkPolicy set to "deny" and policy set to 
```
{"type": "networkPolicyRule",
"action": "allow",
"within": "linked",
}
```

Create a container c1 and link it to another container c2 by setting instanceLinks . (UI does not have support for this)

```
    link_server = client.create_container(name=test_name + '-server',
                                          imageUuid=TEST_IMAGE_UUID,
                                          networkMode=MANAGED_NETWORK,
                                          hostname=test_name + '-server',
                                          environment={
                                              'VALUE': random_val
                                          },
                                          requestedHostId=hosts[2].id)
    link_client = client.create_container(name=test_name + '-client',
                                          imageUuid=TEST_IMAGE_UUID,
                                          networkMode=MANAGED_NETWORK,
                                          ports=['3000:3000'],
                                          hostname=test_name + '-client1',
                                          instanceLinks={
                                              'client': link_server.id
                                          },
                                          requestedHostId=hosts[0].id)
```

Not able to ping c2 fro c1 using its managed ip.

metadata of c1 has "links" which points to uuid of the linked container c2.

```
root@test-631116-client1:/# curl rancher-metadata/latest/self/container/links/client
69526a46-6013-4f31-96b7-178fb407fe25
```

**Comment by sangeethah** [comment]
@leodotcloud , Not sure if we want to support linking of containers using the ```"within": "linked"``` network policy or should this be restricted to service linking only.

**Comment by sangeethah** [comment]
Another use case:
Create a service S1 and link it to a container C1 using instanceLinks set in launch_config:

```
                   "instanceLinks": {
                         link_name:
                             link_container.id},
```

containers from service S1 is not able to reach c1.

**Comment by leodotcloud** [comment]
Fixed in `rancher/network-policy-manager:v0.2.4`

**Comment by sangeethah** [comment]
Tested with rancher-server version - v1.5.0-rc1 and rancher/network-policy-manager:v0.2.4.

Have defaulNetworkPolicy set to "deny" and policy set to
```
{"type": "networkPolicyRule",
"action": "allow",
"within": "linked",
}
```
Scenario1:
Create container c1.
Create container c2 and link it to c1.

From c2 , able to reach c1.
From c1, not able to reach c2.

Scenario2:
Create container c1.
Create  service S1 and link it to c1.

From containers of S1 , able to reach c1.
From c1, not able to reach containers of S2.
